src/test/javascript/spec/component/team/team-participation-table.component.spec.ts

```diff
@@ -182,50 +176,54 @@ describe('TeamParticipationTableComponent', () => {
                 comp = fixture.componentInstance;
                 debugElement = fixture.debugElement;
                 teamService = TestBed.inject(TeamService);
+                router = TestBed.inject(Router);
             });
     });
 
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
     beforeEach(fakeAsync(() => {
-        const exercisesStub = sinon.stub(teamService, 'findCourseWithExercisesAndParticipationsForTeam');
-        exercisesStub.returns(of(new HttpResponse({ body: course })));
+        jest.spyOn(teamService, 'findCourseWithExercisesAndParticipationsForTeam').mockReturnValue(of(new HttpResponse({ body: course })));
         comp.course = course;
         comp.exercise = exercise4;
         comp.team = mockTeam;
+        routerStub = jest.spyOn(router, 'navigate').mockImplementation();
         comp.ngOnInit();
         tick();
     }));
 
     it('Exercises for one team are loaded correctly', fakeAsync(() => {
         // Make sure that all 3 exercises were received for exercise
-        expect(comp.exercises).to.have.length(course.exercises?.length!);
+        expect(comp.exercises).toHaveLength(course.exercises?.length!);
```

Comment:
```suggestion
        expect(comp.exercises).toHaveLength(course.exercises!.length);
```

Reply:
Thanks for pointing this out. I changed it.